projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/SentenceAppointmentService.kt

```diff
@@ -45,10 +44,12 @@ class SentenceAppointmentService(
 
             createAppointment.let {
                 val numberOfAppointments = createAppointment.until?.let {
-                    Period.between(
-                        createAppointment.start.toLocalDate(),
-                        it.toLocalDate()
-                    ).days.div(createAppointment.interval.value)
+                    val duration = Duration.between(
+                        createAppointment.start.toLocalDateTime(),
+                        it.toLocalDateTime()
+                    ).toDays() + 1
+
+                    ceil(duration.toDouble().div(createAppointment.interval.value)).toInt()
```

Comment:
Why do you need the ceil / toDouble here, and the plus 1 above? If you have a duration of 30 days and an interval of 7 days, should there be 4 appointments or 5? I may be misunderstanding, but I'd expect 4 - if so, removing the ceil and the plus 1 gives the right answer.  Example here: https://pl.kotl.in/CeB_t339V
```suggestion
                    val duration = Duration.between(
                        createAppointment.start.toLocalDateTime(),
                        it.toLocalDateTime()
                    ).toDays()
                    (duration / createAppointment.interval.value).toInt()
```

Reply:
it is matching delius functionality.

Reply:
Ah okay makes sense now - because Delius is creating one at the start time as well as at each interval.  Might still be worth simplifying this then, to just add 1 at the end?
```suggestion
                    val duration = Duration.between(
                        createAppointment.start.toLocalDateTime(),
                        it.toLocalDateTime()
                    ).toDays()

                    (duration / createAppointment.interval.value).toInt() + 1
```